src/sentry/data_export/tasks.py

```diff
@@ -113,20 +111,49 @@ def process_issues_by_tag(data_export, file, limit, environment_id):
             # See associated comment in './utils.py'
             gtv_list = convert_to_utf8(gtv_list_unicode)
             if limit and limit < next_offset:
-                # Since the next offset will pass the limit, write the remainder and quit
                 writer.writerows(gtv_list[: limit % SNUBA_MAX_RESULTS])
                 break
             else:
                 writer.writerows(gtv_list)
                 iteration += 1
 
 
-def process_discover(data_export, file):
-    # TODO(Leander): Implement processing for Discover
-    raise NotImplementedError("Discover processing has not been implemented yet")
+def process_discover(data_export, file, limit, environment_id):
+    """
+    Convert the discovery query to a CSV, writing it to the provided file.
+    """
+    try:
+        processor = DiscoverProcessor(
+            discover_query=data_export.query_info, organization_id=data_export.organization_id
+        )
+    except ExportError as error:
+        metrics.incr("dataexport.error", tags={"error": six.text_type(error)}, sample_rate=1.0)
+        logger.info("dataexport.error: {}".format(six.text_type(error)))
+        capture_exception(error)
+        raise error
+
+    writer = create_writer(file, processor.header_fields)
+    iteration = 0
+    with snuba_error_handler(logger=logger):
+        while True:
```

Comment:
Could you please avoid the `while True` and provide a meaningful condition here, so it is less easy to break this and turning it into an infinite loop?

Reply:
I'd actually already used this implementation elsewhere in the file, so I'll try to refactor it in separate ticket for both types of exports `Issues-by-Tag` and these `Discover` ones.

The original reason I'd used a `while True` was because I thought it was easier to read and I have no way of knowing the amount of rows that are going to be in any CSV as they're being assembled.

Reply:
Will request your review on the separate ticket though, thanks for the note

Reply:
@fpacifici I overestimated how hard a refactor would be, I added it in this PR let me know what you think!